v2/handler.go

```diff
@@ -122,7 +128,12 @@ type UnaryEncodingHandler interface {
 	// An error may be returned in case of failures. BadRequestError must be
 	// returned for invalid requests. All other failures are treated as
 	// UnexpectedErrors.
-	Handle(ctx context.Context, req *Request, reqBody interface{}) (*Response, interface{}, error)
+	Handle(ctx context.Context, reqBody interface{}) (interface{}, error)
```

Comment:
Is this the signature we want for transport-level streams? I was under the impression that this would still be performing on `Request` and `Response`, as in
```
Handle(ctx context.Context, req *Request) (*Response, error)
```

It looks like this signature should be for encoding-level handlers, right?

Reply:
Would you entertain making the Response a non-pointer?

Reply:
I'm not necessarily opposed. Just to avoid accidental mutation?